### PR TITLE
driver: sensor: i2c: icm20608: initial support

### PR DESCRIPTION
**Pull request by wuhanstudio**
Add initial support for ICM20608 IMU.

### CODE REVIEW

**Review by teburd — COMMENTED** [review]
Driver looks very nicely done, trigger seems sensibly handled. Some minor questions I have pinged some people to hopefully look at which might require some small changes.

thank you for contributing!

File: drivers/sensor/tdk/icm20608/icm20608.c [hunk 1]
```diff
@@ -0,0 +1,416 @@
...
+static int icm20608_sample_fetch(const struct device *dev, enum sensor_channel chan)
+{
+	struct icm20608_data *drv_data = dev->data;
+	const struct icm20608_config *cfg = dev->config;
+
+	__ASSERT_NO_MSG(chan == SENSOR_CHAN_ALL);
+
+	uint8_t read_buff[ICM20608_SENS_READ_BUFF_LEN];
+
+	if (!icm20608_ready_to_read(dev)) {
```
[the remark this answers is not in the document]
Comment (teburd) [reply drivers/sensor/tdk/icm20608/icm20608.c:179]: Sensor WG discussed 10/7/2024

* Sensor fetch should block until sample is ready and return sample
* If non-blocking behavior is needed then WG recommends moving to read + decode (newer API) which is non-blocking by design.

File: dts/bindings/sensor/invensense,icm20608.yaml [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+      Valid range: 0 - 255
+
+  gyro-dlpf:
+    type: int
+    required: true
+    description: |
+      Default digital low pass filter frequency of gyroscope.
+      Maps to DLPF_CFG field in Configuration setting.
+    enum:
+      - 250
```
[the remark this answers is not in the document]
Comment (wuhanstudios) [reply dts/bindings/sensor/invensense,icm20608.yaml:37]: Here's the register map of ICM-20608-G:
https://invensense.tdk.com/wp-content/uploads/2015/03/RM-000030-v1.0.pdf

[the remark this answers is not in the document]
Comment (wuhanstudio) [reply dts/bindings/sensor/invensense,icm20608.yaml:37]: Here's the register map of ICM-20608-G:
https://invensense.tdk.com/wp-content/uploads/2015/03/RM-000030-v1.0.pdf

**Pull request closed without merging**